File: haushaltsbuch/src/test/java/org/github/haschi/infrastruktur/Anweisungskonfiguration.java
package org.github.haschi.infrastruktur;

import org.axonframework.commandhandling.gateway.CommandGateway;
import org.axonframework.config.Configuration;
import org.axonframework.config.DefaultConfigurer;
import org.github.haschi.haushaltsbuch.modell.Haushaltsbuch;
import org.github.haschi.haushaltsbuch.modell.Inventur;
import org.picocontainer.Startable;

public class Anweisungskonfiguration implements Startable
{

    public CommandGateway commandGateway()
    {
        return konfiguration.commandGateway();
    }

    public Configuration konfiguration()
    {
        return konfiguration;
    }

    private final Configuration konfiguration;

    public Anweisungskonfiguration(final EventStoreLieferant storagelieferant)
    {
        this.konfiguration = DefaultConfigurer.defaultConfiguration()
                .configureAggregate(Inventur.class)
                .configureAggregate(Haushaltsbuch.class)
                .configureEventStore(storagelieferant::eventBus)
                .buildConfiguration();
    }

    @Override
    public void start()
    {
        konfiguration.start();
    }

    @Override
    public void stop()
    {
        konfiguration.shutdown();
    }
}

File: haushaltsbuch-domain/src/test/kotlin/org/github/haschi/infrastruktur/Anweisungskonfiguration.kt
package org.github.haschi.infrastruktur

import org.axonframework.commandhandling.gateway.CommandGateway
import org.axonframework.config.Configuration
import org.axonframework.config.DefaultConfigurer
import org.github.haschi.domain.haushaltsbuch.modell.Haushaltsbuch
import org.github.haschi.domain.haushaltsbuch.modell.Inventur
import org.picocontainer.Startable

class Anweisungskonfiguration(storagelieferant: EventStoreLieferant) : Startable
{

    fun commandGateway(): CommandGateway
    {
        return konfiguration.commandGateway()
    }

    fun konfiguration(): Configuration
    {
        return konfiguration
    }

    private val konfiguration: Configuration = DefaultConfigurer.defaultConfiguration()
            .configureAggregate(Inventur::class.java)
            .configureAggregate(Haushaltsbuch::class.java)
            .configureEventStore({ storagelieferant.eventBus(it) })
            .buildConfiguration()

    override fun start()
    {
        konfiguration.start()
    }

    override fun stop()
    {
        konfiguration.shutdown()
    }
}
